Sources/MongoSwift/BSON/Document+Sequence.swift

```diff
@@ -256,18 +256,11 @@ public class DocumentIterator: IteratorProtocol {
     /// - Throws:
     ///   - `RuntimeError.internalError` if the current value of this `DocumentIterator` cannot be decoded to BSON.
     internal func safeCurrentValue() throws -> BSONValue {
-        switch self.currentType {
-        case .symbol:
-            return try Symbol.asString(from: self)
-        case .dbPointer:
-            return try DBPointer.asDocument(from: self)
-        default:
-            guard let curVal = try DocumentIterator.bsonTypeMap[currentType]?.from(iterator: self) else {
-                throw RuntimeError.internalError(message: "Unknown BSONType for iterator's current value.")
-            }
-
-            return curVal
+        guard let curVal = try DocumentIterator.bsonTypeMap[currentType]?.from(iterator: self) else {
```

Comment:
this looks a little magical, could we split it up into extracting the type (and throwing if the type is not found), and the actual type instantiation?